### PR TITLE
HBHE: add photostatitics uncertainty in the M2

### PR DESCRIPTION
**Pull request by mariadalfonso**
This PR is a follow up on the #17311 
And make use of the new GT https://github.com/cms-sw/cmssw/pull/17530

The noise uncertainty now is added in the M2 chi2 also for the HPD.
validation slides are here
http://dalfonso.web.cern.ch/dalfonso/PhotoStatHPD.pdf

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

there appear to be run time issues with #17530
Also, #17530 GT updates geometry in 2017, which makes this PR change incrementally untestable.

So, this PR may not move until #17530 is merged.

**Comment by franzoni** [comment]
Alca signature subject to resolution of https://github.com/cms-sw/cmssw/pull/17530
Where investigations are on going re content of payloads

**Comment by abdoulline** [comment]
I've fixed the content of HcalSiPMParameters for Run 1, Walter will submit a new tag (HcalSiPMParameters_v1.2_mc) today

File: RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc [hunk 1]
```diff
@@ -557,23 +564,15 @@ void PulseShapeFitOOTPileupCorrection::phase1Apply(const HBHEChannelInfo& channe
...
-    if(!channelData.hasTimeInfo() && (charge-ped)>channelData.tsPedestalWidth(ip)) {
-
-      // Note1. the fcByPE is not in the DB, but is hard coded in python
-      // https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_0/SimCalorimetry/HcalSimProducers/python/hcalSimParameters_cfi.py#L62
-
-      // Note2. (from kPedro): the output number of photoelectrons after smearing is treated very differently for SiPMs: *each* pe is assigned a different time based on a random generation from the Y11 pulse plus the SimHit time. In HPDs, the overall pulse is shaped all at once using just the SimHit time.
-
-      noisePHArr[ip] = sqrt((charge-ped)*0.3305);
+    if(!channelData.hasTimeInfo() && (charge-ped)>noise_) {
+      noisePHArr[ip] = sqrt((charge-ped)*channelData.fcByPE());
```
Comment (slava77) [inline RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc]: is the value here consistent with 0.3305 used in PulseShapeFitOOTPileupCorrection::apply?
I was actually expecting that the same source is used in all cases.

Comment (mariadalfonso) [inline RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc]: with the new GT, for HPD,  printing on the screen the channelData.fcByPE() is 0.3305

Comment (slava77) [inline RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc]: Nice, thank you for the confirmation.
Can we avoid the hardcoded value in the PulseShapeFitOOTPileupCorrection::apply and get it from DB?

Comment (mariadalfonso) [inline RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc]: the amazing class HBHEChannelInfo channelData.fcByPE() is only available for the Phase1Code.

For the run1/run2 code I will likely have to do a lot of gymnastic.
I tried to do this in the past unsuccessfully for the other parameters i.e. the noise https://github.com/mariadalfonso/cmssw/commit/c923f24e48c922581aeee84422af9e8db29cb466
The old M2 code is anyway full of all hard coded numbers so one more or less should not make a big difference. I can see if I can find an easy way, but no promises (maybe we can put in the post-pre5 SW rework)?

**Comment by slava77** [comment]
On 2/17/17 11:54 AM, mariadalfonso wrote:
> (maybe we can put in the post-pre5 SW rework)?

yes, OK.

**Comment by slava77** [comment]
there is a merge conflict here now

**Comment by mariadalfonso** [comment]
@slava77 
I cherry picked giovanni's commit, so that now is w/o conflicts

**Comment by franzoni** [comment]
please test

**Comment by slava77** [comment]
On 2/19/17 12:51 AM, mariadalfonso wrote:
> @slava77 <https://github.com/slava77>
> I cherry picked giovanni's commit, so that now is w/o conflicts

At this point  (the GT PR is merged) it's technically better to remove
edits in autoCond.py from this PR.

File: RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc (hunk 1 above)
Comment (slava77) [inline RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc]: @igv4321  please clarify what is the status of moving all of the hbhe reco to the phase-1  code?

Comment (igv4321) [inline RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc]: We are working on it. Approximate time line when it happens is end of Feb. @halilg might be able to provide more details.

**Comment by mariadalfonso** [comment]
@slava77 
rebased on top of CMSSW_9_0_X_2017-02-19-1100

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
[the following is based on https://github.com/cms-sw/cmssw/pull/17533	2dae29d; made in CMSSW_9_0_X_2017-02-16-1100 as a baseline; I made a quick check with the later version to confirm that the conclusions do not change]

The main change from this PR is improved chi2 in hbhe hits
10224:
barrel
all_sign860vsorig_ttbar13tevpu2017wf10224p0c_log10hbherechitssorted_hbhereco__reco_obj_obj_chi210

endcap
all_sign860vsorig_ttbar13tevpu2017wf10224p0c_log10hbherechitssorted_hbhereco__reco_obj_obj_chi220

There is also a significant reduction (by x2 in the barrel) in the number of hits with the 3-pulse fit flag (HBHEPulseFitBit=29)

The fraction of non-zero (E>1 MeV) hits is up in the PU sample:
by 7% in the barrel
all_sign860vsorig_ttbar13tevpu2017wf10224p0c_log10hbherechitssorted_hbhereco__reco_obj_obj_energy3

and by 13% in the endcaps
all_sign860vsorig_ttbar13tevpu2017wf10224p0c_log10hbherechitssorted_hbhereco__reco_obj_obj_energy13
Since the 1-pulse fit is tried first, we should expect more OOT PU energy to come in for the fraction of previously 3-pulse fitted hits, which OTOH also had a fraction of in-time energy "eaten" by the OOT component.

The number of neutral hadrons is up by 12%
all_sign860vsorig_ttbar13tevpu2017wf10224p0c_log10recopfcandidates_particleflow__reco_obj_pt73

This transfers to probably 1-2% increase in lower-pt jet response 
wf10224_ak4calo_corovergen_eta_e

Jet samples without pileup show smaller differences in response, with sub-% increase in the TeV range.
Time of rechits is more flat now [wf 10071 flat-pt QCD sample]
wf10071_hittime_vs_e_hb
wf10071_hittime_vs_e_he

chi2 is smaller, as expected
wf10071_log10chi2_vs_e_hb
wf10071_log10chi2_vs_e_he


In data, changes look fairly similar
e.g. in 2016E JetHT (136.761)
chi2 in endcap
all_sign860vsorig_runjetht2016ewf136p761c_log10hbherechitssorted_hbhereco__rereco_obj_obj_chi220
hit energy E>1MeV in endcap
all_sign860vsorig_runjetht2016ewf136p761c_log10hbherechitssorted_hbhereco__rereco_obj_obj_energy13


On the technical side, in PU35 ttbar (wf10224) 
- the hbheprereco time is down by 24% (or about 1% of total reco time)
- given the increase in the number of hits with E>0 some module times are up: 
    - particleFlowClusterHBHE up by 50% (from 7 to 11 ms/evt)
    - particleFlowClusterHCAL up by 15% (negligible for the total; under 2ms/evt)

Overall, this seems ready to sign; I'll wait for jenkins tests to finish.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/17533	fd3b827
- changes are as described: to add the photostatistics component to the error used in hit fit minimization in M2; affects all simulations with HPDs
- jenkins tests pass and comparisons with baseline show differences that start from hbhe hit reco and propagate downstream; differences are consistent with what's found in the local tests summarized in https://github.com/cms-sw/cmssw/pull/17533#issuecomment-280945324
    - the tests after rebase also do not show any irrelevant differences as present earlier in the comparisons due to the GT changes that were included with this PR

**Comment by davidlange6** [comment]
+1

**Pull request merged**